### PR TITLE
Delete dead code

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

**Comment by callunity** [comment]
Since there is no ticket and our JIRA is not externally visible anyway, could you detail some explanation of why we are making this change?

**Comment by morgantaschuk** [comment]
Yes, please explain the Plates/Boxes rationale. Thanks.

**Comment by apmasell** [comment]
I've amended the commit to explain what's going on.

**Comment by tonydebat** [comment]
After merging these changes I get two unit test error:
```
Tests in error:
  testListByProjectNone(uk.ac.bbsrc.tgac.miso.sqlstore.SQLPoolDAOTest)
  testListByProjectId(uk.ac.bbsrc.tgac.miso.sqlstore.SQLPoolDAOTest)
```

**Comment by apmasell** [comment]
Fixed.

**Comment by tonydebat** [comment]
Unit test fix worked. Now I'm getting a flyway migration error:

```
[INFO] Migrating schema `miso` to version 8000 - container constraints
[INFO] Migrating schema `miso` to version 8800 - delete plates
[INFO] Successfully applied 2 migrations to schema `miso` (execution time 00:00.243s).
[INFO] Executing SQL callback: afterMigrate
[WARNING] DB: Trigger does not exist (SQL State: HY000 - Error Code: 1360)
[WARNING] DB: Trigger does not exist (SQL State: HY000 - Error Code: 1360)
[WARNING] DB: Table 'miso.plate' doesn't exist (SQL State: 42S02 - Error Code: 1146)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.004 s
[INFO] Finished at: 2016-10-07T12:11:58-04:00
[INFO] Final Memory: 20M/313M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:3.2.1:migrate (default-cli) on project sqlstore: org.flywaydb.core.internal.dbsupport.FlywaySqlScriptException:
[ERROR] Migration afterMigrate.sql failed
[ERROR] ---------------------------------
[ERROR] SQL State  : 42S02
[ERROR] Error Code : 1146
[ERROR] Message    : Table 'miso.plate' doesn't exist
[ERROR] Location   : db/migration/afterMigrate.sql (/Users/tdebat/code/gsi/miso-lims/sqlstore/target/classes/db/migration/afterMigrate.sql)
[ERROR] Line       : 252
[ERROR] Statement  : CREATE TRIGGER PlateChange BEFORE UPDATE ON Plate
[ERROR] FOR EACH ROW
[ERROR] BEGIN
[ERROR] DECLARE log_message varchar(500) CHARACTER SET utf8;
[ERROR] SET log_message = CONCAT_WS(', ',
[ERROR] CASE WHEN NEW.plateMaterialType <> OLD.plateMaterialType THEN CONCAT('plate material: ', OLD.plateMaterialType, ' → ', NEW.plateMaterialType) END,
[ERROR] CASE WHEN (NEW.locationBarcode IS NULL) <> (OLD.locationBarcode IS NULL) OR NEW.locationBarcode <> OLD.locationBarcode THEN CONCAT('location barcode: ', COALESCE(OLD.locationBarcode, 'n/a'), ' → ', COALESCE(NEW.locationBarcode, 'n/a')) END,
[ERROR] CASE WHEN NEW.size <> OLD.size THEN CONCAT('size: ', OLD.size, ' → ', NEW.size) END,
[ERROR] CASE WHEN (NEW.indexId IS NULL) <> (OLD.indexId IS NULL) OR NEW.indexId <> OLD.indexId THEN CONCAT('index: ', COALESCE(OLD.indexId, 'n/a'), ' → ', COALESCE(NEW.indexId, 'n/a')) END,
[ERROR] CASE WHEN NEW.description <> OLD.description THEN CONCAT('description: ', OLD.description, ' → ', NEW.description) END);
[ERROR] IF log_message IS NOT NULL AND log_message <> '' THEN
[ERROR] INSERT INTO PlateChangeLog(plateId, columnsChanged, userId, message) VALUES (
[ERROR] NEW.plateId,
[ERROR] COALESCE(CONCAT_WS(',',
[ERROR] CASE WHEN NEW.plateMaterialType <> OLD.plateMaterialType THEN 'plateMaterialType' END,
[ERROR] CASE WHEN (NEW.locationBarcode IS NULL) <> (OLD.locationBarcode IS NULL) OR NEW.locationBarcode <> OLD.locationBarcode THEN 'locationBarcode' END,
[ERROR] CASE WHEN NEW.size <> OLD.size THEN 'size' END,
[ERROR] CASE WHEN (NEW.indexId IS NULL) <> (OLD.indexId IS NULL) OR NEW.indexId <> OLD.indexId THEN 'indexId' END,
[ERROR] CASE WHEN NEW.description <> OLD.description THEN 'description' END), ''),
[ERROR] NEW.lastModifier,
[ERROR] log_message);
[ERROR] END IF;
[ERROR] END
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Comment by apmasell** [comment]
Fixed. I forgot about `afterMigrate`.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractPool.java [hunk 1]
```diff
@@ -412,30 +412,16 @@ public boolean equals(Object obj) {
     if (obj == this) return true;
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractPool.java]: line 190 (Javadoc) contains a reference to plates

**Comment by apmasell** [comment]
I've fixed the JavaDocs and some of the code. There's still a plate receipt form where samples arrive on a plate. That's what your seeing in the project-related stuff.

**Pull request merged**